src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -1714,6 +1716,510 @@ public sealed class NullableAttribute : Attribute { }
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F23").TypeWithAnnotations, new byte[] { 1, 1 }, new byte[] { 1, 1 }, "U![]!");
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F24").TypeWithAnnotations, new byte[] { 1, 1 }, new byte[] { 1, 1 }, "C<U!>!");
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F25").TypeWithAnnotations, new byte[] { 0, 1 }, new byte[] { 0, 1 }, "S<U!>");
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F31").TypeWithAnnotations, new byte[] { 0 }, new byte[] { 0 }, "V");
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F32").TypeWithAnnotations, new byte[] { 0, 0 }, new byte[] { 0 }, "V?");
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F33").TypeWithAnnotations, new byte[] { 1, 0 }, new byte[] { 1, 0 }, "V[]!");
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F34").TypeWithAnnotations, new byte[] { 1, 0 }, new byte[] { 1 , 0}, "C<V>!");
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F35").TypeWithAnnotations, new byte[] { 0, 0 }, new byte[] { 0, 0 }, "S<V>");
+            }
+        }
+
+        private static readonly SymbolDisplayFormat _displayFormat = SymbolDisplayFormat.TestFormat.
+            WithMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier | SymbolDisplayMiscellaneousOptions.UseSpecialTypes).
+            WithCompilerInternalOptions(SymbolDisplayCompilerInternalOptions.IncludeNonNullableTypeModifier);
+
+        private static void VerifyBytes(TypeWithAnnotations type, byte[] expectedPreviously, byte[] expectedNow, string expectedDisplay)
+        {
+            var builder = ArrayBuilder<byte>.GetInstance();
+            type.AddNullableTransforms(builder);
+            var actualBytes = builder.ToImmutableAndFree();
+
+            Assert.Equal(expectedNow, actualBytes);
+            Assert.Equal(expectedDisplay, type.ToDisplayString(_displayFormat));
+
+            var underlyingType = type.SetUnknownNullabilityForReferenceTypes();
+
+            // Verify re-applying the same bytes gives the same result.
+            TypeWithAnnotations updated;
+            int position = 0;
+            Assert.True(underlyingType.ApplyNullableTransforms(0, actualBytes, ref position, out updated));
+            Assert.True(updated.Equals(type, TypeCompareKind.ConsiderEverything));
+
+            // If the expected byte[] was different with earlier builds, verify that
+            // applying the previous byte[] does not succeed.
+            if (!expectedPreviously.SequenceEqual(expectedNow))
+            {
+                position = 0;
+                Assert.True(underlyingType.ApplyNullableTransforms(0, ImmutableArray.Create(expectedPreviously), ref position, out updated) || position < expectedPreviously.Length);
```

Comment:
>underlyingType.ApplyNullableTransforms(0, ImmutableArray.Create(expectedPreviously), ref position, out updated) [](start = 28, length = 111)

Should this condition be negated?

Reply:
Or perhaps removed?

---
In reply to: [294947325](https://github.com/dotnet/roslyn/pull/36519#discussion_r294947325) [](ancestors = 294947325)